packages/rocketchat-integrations/server/processWebhookMessage.js

```diff
@@ -1,10 +1,35 @@
 import _ from 'underscore';
 import s from 'underscore.string';
+import moment from 'moment';
 
 this.processWebhookMessage = function(messageObj, user, defaultValues = { channel: '', alias: '', avatar: '', emoji: '' }, mustBeJoined = false) {
 	const sentData = [];
 	const channels = [].concat(messageObj.channel || messageObj.roomId || defaultValues.channel);
-
+	if (!user._id) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+			method: 'processWebhookMessage'
+		});
+	}
+	if (messageObj.ts) {
+		const tsDiff = Math.abs(moment(messageObj.ts).diff());
+		if (tsDiff > 60000) {
+			throw new Meteor.Error('error-message-ts-out-of-sync', 'Message timestamp is out of sync', {
+				method: 'sendMessage',
+				message_ts: messageObj.ts,
+				server_ts: new Date().getTime()
+			});
+		} else if (tsDiff > 10000) {
+			messageObj.ts = new Date();
+		}
+	} else {
+		messageObj.ts = new Date();
+	}
+	if ((messageObj.msg && messageObj.msg.length > RocketChat.settings.get('Message_MaxAllowedSize')) ||
```

Comment:
I worry about this one. I have a feeling we might break integrations which rely on getting around the `Message_MaxAllowedSize` for integrations...

Reply:
@graywolf336 Do you think integrations should bypass the server rules?

Reply:
To be honest, I'm conflicted about that. I almost would like us to get feedback from the community regarding it. But things like if the user is muted should be respected..